### PR TITLE
customizing keypoint compression

### PR DESCRIPTION
**Pull request by vibhaa**

### CODE REVIEW

File: src/aiortc/codecs/keypointcodec.py [hunk 1]
```diff
@@ -112,10 +116,14 @@ def custom_encode(keypoint_dict):
     binary_str = ""
     num_bins = 2**(NUM_JACOBIAN_BITS - 1)
```
Comment (pkarimib) [inline src/aiortc/codecs/keypointcodec.py:117]: space?

File: src/aiortc/codecs/keypointcodec.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 SCALE_FACTOR = 256//2
 NUM_KP = 10
-NUM_JACOBIAN_BITS = int(os.environ.get('JACOBIAN_BITS', 6))
+NUM_JACOBIAN_BITS = int(os.environ.get('JACOBIAN_BITS', -1))
```
Comment (pkarimib) [inline src/aiortc/codecs/keypointcodec.py:15]: Is 'JACOBIAN_BITS' a list? If not, what does it mean to have -1 for the default value?

Comment (pkarimib) [reply src/aiortc/codecs/keypointcodec.py:15]: oh got it, nevermind

File: src/aiortc/codecs/keypointcodec.py [hunk 3]
```diff
@@ -192,8 +192,16 @@ def _convert(data: bytes, width: int) -> bytes:
...
-        keypoint_dict['pts'] = 5
+
+        if NUM_JACOBIAN_BITS == -1:
+            keypoint_info_struct = KeypointInfo()
+            keypoint_info_struct.ParseFromString(keypoint_str)
+            assert(keypoint_info_struct.IsInitialized())
+            keypoint_dict = keypoint_struct_to_dict(keypoint_info_struct)
+        else:
+            keypoint_dict = custom_decode(keypoint_str)
+            keypoint_dict['pts'] = 5
```
Comment (pkarimib) [inline src/aiortc/codecs/keypointcodec.py:203]: Why 5?

Comment (pkarimib) [reply src/aiortc/codecs/keypointcodec.py:203]: Can we have a global constant?

**Review by pkarimib — CHANGES_REQUESTED** [review]
Mostly questions, minor comments

File: src/aiortc/codecs/keypointcodec.py [hunk 4]
```diff
@@ -63,6 +63,115 @@ def keypoint_struct_to_dict(keypoint_info_struct):
...
+
+""" compute the bin corresponding to the jacobian value
+    based on the Huffman dictionary for the desired
+    number of bins/bits
+"""
+def jacobian_to_bin(value, num_bins):
+    sign = int(value > 0)
+    value = abs(value)
+
+    if value > 3:
```
[the remark this answers is not in the document]
Comment (pkarimib) [reply src/aiortc/codecs/keypointcodec.py:81]: Do we actually know the ranges for all values that we picked 3? is it based on observations?

[the remark this answers is not in the document]
Comment (vibhaa) [reply src/aiortc/codecs/keypointcodec.py:81]: largely based on observations - hard to do away without hard-coded numbers here

File: src/aiortc/codecs/keypointcodec.py (hunk 3 above)
Comment (vibhaa) [reply src/aiortc/codecs/keypointcodec.py:203]: Good point - This is just a hack for now to set it to something explicit - it doesn't get used, so I decided to make it something random

File: src/aiortc/codecs/keypointcodec.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (pkarimib) [reply src/aiortc/codecs/keypointcodec.py:82]: okay, we can let it slide, but I'm not really sure

**Review by pkarimib — APPROVED** [review]

**Pull request merged**